[mingyang143] iP

mingyang143 opened the pull request:
# Friday

> The only application that you need!

Friday makes you feel like its Friday everyday as it:

- keeps track of all your tasks to give you a peace of mind
- easy to use
- ~~fast~~ **SUPER FAST** to learn

All you need to do is:

1. Download the CLI executable file from [here](https://github.com/mingyang143/ip/releases/download/A-JUnit/friday.jar)
2. Open up the terminal and navigate to the project directory
3. Run "java -jar" command
4. Add your tasks and look at the magic🪄🔥

And it is completely **FREE!**

Features:

- [x] Managing tasks, deadlines and event
- [x] Task Finding
- [x] Task saving to local file in computer

If you wanna take a look at a snippet of the code, here is the `main` method: 

```java
public static void main(String[] args) {
        new Friday("data/fridayTaskList.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mingyang143 committed:
added Friday.java

mingyang143 committed:
updated Friday.java with interactive UI

mingyang143 committed:
Added list for Friday.java

mingyang143 committed:
Mark as Done

mingyang143 committed:
added Deadline, Event and Todo files

mingyang143 committed:
Added TextUiTesting

mingyang143 committed:
Added error handling

mingyang143 committed:
added delete functionality and fixed bugs

mingyang143 committed:
Refactored code to use enum and switch statements

Commit:
Add read and write functionality to Friday

Commit:
Use LocalDate intead of String to represent date

Commit:
Refactor code to use more oop

The code in the main function of Friday.java is very messy due to a lack
of encapsulation of related code.

The code in Friday.java is very unreadable.

Extracting the code for Ui, Command, Parser, Storage and TaskList into
their respective classes.

Encapsulating the code into those respective classes improves
readability.

Commit:
organize the classes into packages

Commit:
Merge branch 'add-gradle-support'

Commit:
add gradle to project

Commit:
Revert "add gradle to project"

This reverts commit 2ba110af2f05717a26a27bd473ea6714b39214d4.

Commit:
Add gradle to project

Commit:
Add JUnit testing to Todo.java and Storage.java

Commit:
Package app as jar file

Commit:
Add Javadoc comments for the entire application

Commit:
Modify code to follow coding standard

Commit:
Modify code to follow coding standard

Commit:
Add find command

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/fridayTaskList.txt
#	src/main/java/parser/Parser.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/fridayTaskList.txt
#	src/main/java/parser/Parser.java

shizy commented on src/main/java/app/Friday.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isEndScanner = false;
```
I like that you followed naming conventions for booleans!

shizy commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,50 @@
+            if (taskIndex < 0 || taskIndex >= tasks.size()) {
+                throw new InvalidDeleteArgument();
+            }
+            Task taskToDelete = tasks.getTask(taskIndex);  // Get the task to be deleted
+            tasks.removeTask(taskIndex);  // Remove the task from the list
+            ui.showTaskRemoved(taskToDelete, tasks.size());  // Show the user that the task was removed
+
+            storage.save(tasks.getTasks());  // Save the updated task list to the file
```
I like the use of trailing comments to explain what is happening at those lines, it makes following the code much easier 👍

shizy commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,50 @@
+package command;
+
+import fridayException.FridayException;
+import fridayException.InvalidDeleteArgument;
```
I like that you specified the individual exception classes to be imported, instead of simply importing * from the fridayException package!

shizy commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,96 @@
+package parser;
+
+import command.AddCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.SearchCommand;
+import command.UnmarkCommand;
+import fridayException.FridayException;
+import fridayException.InvalidDeadlineArgument;
+import fridayException.InvalidDeleteArgument;
+import fridayException.InvalidEventArgument;
+import fridayException.InvalidFindArgument;
+import fridayException.InvalidFridayCommand;
+import fridayException.InvalidMarkArgument;
+import fridayException.InvalidSearchArgument;
+import fridayException.InvalidTodoArgument;
+import fridayException.InvalidUnmarkArgument;
+import task.Deadline;
+import task.Event;
+import task.ToDo;
```
Well done with importing the specific classes you needed, instead of just importing * from these packages! 👍

shizy commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,96 @@
+        String[] parts = userInput.split(" ");
+        String command = parts[0].toLowerCase();
+
+        switch (command) {
```
It's great that you abided by the indentation format for switch case statements

shizy commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,75 @@
+    @Override
+    public String writeToFile() {
+        return "E " + (super.isCompleted() ? "0" : "1") + " "
+                + this.getTaskName() + "/from " + this.startEvent + " /to " + this.endEvent;
```
I like that you followed the indentation format for wrapped lines 👍

shizy commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,136 @@
+     * @return True if the file is not corrupted, false otherwise.
+     */
+    public boolean isFileUncorrupted(File file) {
+        Pattern TODO_PATTERN = Pattern.compile("^T\\s+\\d\\s+\\w+.*$");
+        Pattern EVENT_PATTERN = Pattern.compile("^E\\s+\\d\\s+\\w+.*\\s+/from\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)\\s+/to\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)$");
+
+        Pattern DEADLINE_PATTERN = Pattern.compile("^D\\s+\\d\\s+\\w+.*\\s+/by\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)$");
```
Great to see that constants (for this purpose) are named with all uppercase 👍

mingyang143 replied:
Thank you for spending your time to review my code!

shizy reviewed (commented):
Overall, I don't really see any clear coding standard violations, with formatting being adhered to very well.
LGTM 💯

SKarthikeyan28 commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,50 @@
+            }
+            Task taskToDelete = tasks.getTask(taskIndex);  // Get the task to be deleted
+            tasks.removeTask(taskIndex);  // Remove the task from the list
+            ui.showTaskRemoved(taskToDelete, tasks.size());  // Show the user that the task was removed
```
I like the naming of variables as it is very self explanatory on what is happening

SKarthikeyan28 commented on src/main/java/command/SearchCommand.java:
```diff
@@ -0,0 +1,35 @@
+     *
+     * @param searchDate The date to search for tasks on.
+     */
+    public  SearchCommand(LocalDate searchDate) {
```
I think there might be an extra whitespace between public and SearchCommand

SKarthikeyan28 commented on src/main/java/fridayException/InvalidDeadlineArgument.java:
```diff
@@ -0,0 +1,11 @@
+public class InvalidDeadlineArgument extends FridayException {
+    public InvalidDeadlineArgument() {
+        super("The description of a deadline and date to be done by cannot be empty. " + "\n"
+                + "     Please enter a valid deadline command: deadline <description> /by <date>.");
```
Good job on having breaks to improve readability

SKarthikeyan28 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    public static Command parse(String userInput) throws FridayException {
+        String[] parts = userInput.split(" ");
+        String command = parts[0].toLowerCase();
```
Since command is also an abstract class, maybe can use a different name or edit the variable name?

SKarthikeyan28 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+     *
+     * @return The number of tasks in the task list.
+     */
+    public int size() {
```
Similar to above, I think can slightly change the name to make it sound like a verb like "getSize"

SKarthikeyan28 commented on src/test/java/storage/StorageTest.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    @Test
+    public void isFileUncorrupted_validFile_trueReturned() throws IOException {
```
Naming of test methods are good, very self explanatory

SKarthikeyan28 reviewed (commented):
Overall I found your code easy to read and understand. The code in the class files also adhere to java coding style. **LGTM**!!

Commit:
Add checkstyle config files

Commit:
Add Javafx GUI

Commit:
Use Varargs in DialogBox.java

Commit:
Add Javadoc to all public methods and classes

Currently, the codebase lacks detailed documentation for public methods
and classes. This makes it difficult for developers to understand the
purpose, parameters, and expected behavior of various parts of the code.

Lack of documentation can lead to misinterpretations and errors,
especially as the project grows and more developers contribute.
Adding Javadoc will improve code readability and maintainability.

Add Javadoc to all public methods and classes.

Javadoc is added to facilitate understanding of current and future
contributors, ensuring that the code is self-explanatory and reducing
the likelihood of bugs caused by misunderstanding.
This also aligns with best practices in Java development.

Commit:
Add assertions to ensure code validity

Currently, the Friday application lacks internal checks that validate
assumptions about the state of important variables and inputs.
This can lead to unexpected behavior and make debugging difficult.

Without assertions, key assumptions about objects such as user input,
UI and storage components are not verified, increasing the risk of
NullPointerException and incorrect states.

Add assertions throughout the Friday application code to verify
non-null inputs, valid file paths, and proper initialization of UI
elements, storage, and parser. These assertions ensure that the
application handles unexpected states more gracefully.

The assertions are added this way to catch potential issues during
development while maintaining performance in production, as assertions
are typically disabled in runtime environments.

mingyang143 committed:
Merge pull request #1 from mingyang143/branch-A-Assertions

Add assertions to ensure code validity

Commit:
Improve code quality by enhancing method naming

Several method names do not clearly convey their purpose,
making the codebase harder to read and maintain.

This affects the overall readability and may lead to confusion when
understanding the code.

Clear and descriptive method names are essential for improving code
clarity and maintainability. By refining the method names,
future developers will have a clearer understanding of the code’s
functionality at a glance.

Renamed methods to reflect their purpose better, aligning with naming
conventions and best practices. This makes the codebase more intuitive,
aiding both development and debugging efforts.

mingyang143 committed:
Merge branch 'master' into branch-A-CodeQuality

mingyang143 committed:
Merge pull request #2 from mingyang143/branch-A-CodeQuality

Improve code quality by enhancing method naming

Commit:
Refactor file loading to use Java streams

Currently, the task loading logic relies on an imperative loop to read
and process each line of the file.

Using Java streams simplifies the process by leveraging functional
programming constructs, reducing the complexity of the code while
making it more concise and easier to maintain.

Refactor the loadTasksFromFile method to use streams for reading lines,
mapping them to tasks, filtering out null values, and collecting the
tasks into a list.

This improves code readability and aligns with modern Java practices.

mingyang143 committed:
Merge pull request #3 from mingyang143/branch-A-Streams

Refactor file loading to use Java streams

Commit:
Add gradle.yml for GitHub Actions CI workflow

Commit:
Add sort command for deadlines based on the earliest date

Currently, the deadlines are not sorted, which makes it harder to
prioritize tasks and manage upcoming deadlines effectively.

Unsorted deadlines can lead to confusion when handling tasks with
urgent due dates, as it is difficult for users to identify which tasks
require immediate attention.

Add a sort command that arranges deadlines in ascending order based on
their due date. Use a comparator to compare LocalDate values from
the getDeadline() method.

This change ensures that deadlines are displayed in the correct
chronological order, allowing users to better manage and prioritize
their tasks. Sorting deadlines by urgency improves task planning and
overall user experience.

mingyang143 committed:
Improve GUI

mingyang143 committed:
Add unique personality to Friday

mingyang143 committed:
Add more error handling

mingyang143 committed:
Add Ui.png

mingyang143 committed:
Add Ui.png

mingyang143 committed:
Update README.md

mingyang143 committed:
Update README.md

mingyang143 committed:
Add testing for Parser.java

mingyang143 committed:
Update workflows

mingyang143 committed:
Update all Java files to comply with Checkstyle conventions

mingyang143 committed:
Edit runtest.sh and runtest.bat

mingyang143 committed:
Update runtest.bat and runtest.sh

mingyang143 committed:
Update gradle.yml

mingyang143 committed:
Update gradle.yml

mingyang143 committed:
Update gradle.yml

mingyang143 committed:
Update gradle.yml

mingyang143 committed:
Update gradle.yml

mingyang143 committed:
Update EXPECTED.TXT

mingyang143 committed:
Update test-ui-test

mingyang143 committed:
Edit gradle.yml

mingyang143 committed:
Update README.md

mingyang143 committed:
Update README.md

mingyang143 committed:
Update InvalidFridayCommand.java

mingyang143 committed:
Update InvalidFridayException.java

mingyang143 committed:
Remove dead code from the codebase

mingyang143 committed:
Update README.md

mingyang143 committed:
Update README.md
